misk-crypto/src/main/kotlin/misk/crypto/Cipher.kt

```diff
@@ -0,0 +1,49 @@
+package misk.crypto
+
+import com.google.crypto.tink.Aead
+import com.google.crypto.tink.JsonKeysetWriter
+import com.google.crypto.tink.KeysetHandle
+import com.google.crypto.tink.aead.AeadFactory
+import java.io.ByteArrayOutputStream
+import java.util.Base64
+
+/**
+ * An even simpler interface than Tink's Aead.
+ * This also makes it easier to mock Ciphers for testing purposes.
+ */
+interface Cipher {
+  fun encrypt(plaintext: ByteArray?): ByteArray?
+  fun decrypt(ciphertext: ByteArray?): ByteArray?
+  fun keyInfo(): Map<String, String>
```

Comment:
could we document this method?

Reply:
as well as defining what keys are set? is returning a more explicit object with tink and encryptedKey fields more clear here?

Reply:
nit: does this need to be a method or should it be a `val`?

Reply:
@wesleyk the `Map<String, String>` was basically just an experiment here TBH.
I plan to use this value later as  the output of an HTTP endpoint, and I'm assuming I'll add more fields to it later.
That why a map was a pretty comfortable choice here (though naive).

WDYT about putting this in a data class?

Reply:
yeah I'd be for using an actual data class.

Reply:
moshi makes it easy to then expose that data class over an API if you so desire